### PR TITLE
Add inifintely

### LINKED ISSUE #327: Could `forever` be `Applicative f => f a -> Void` instead?
Just asking, since

    foo :: IO (Async ())
    foo = forever $ do ...

vs.

    bar :: IO (Async ())
    bar = async $ forever $ do ...

seems like common error for less attentive guys like me.

### PR DESCRIPTION
**Pull request by aleator**
Resolves #327

### HLint

- [X] I've changed the exposed interface (add new reexports, remove reexports, rename reexported things, etc.).
  - [X] I've updated [`hlint.dhall`](https://github.com/kowainik/relude/blob/master/hlint/hlint.dhall) accordingly to my changes (add new rules for the new imports, remove old ones, when they are outdated, etc.).
  - [ ] I've generated the new `.hlint.yaml` file (see [this instructions](https://github.com/kowainik/relude#generating-hlintyaml)).

### General

- [X] I've updated the [CHANGELOG](https://github.com/kowainik/relude/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [X] All new and existing tests pass.
- [X] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [ ] I've used the [`stylish-haskell` file](https://github.com/kowainik/relude/blob/master/.stylish-haskell.yaml).
- [X] My change requires the documentation updates.
  - [X] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 `relude` uses [PVP Versioning][1].
 The changelog is available [on GitHub][2].
 
+## 0.X.0
```
Comment (chshersh) [inline CHANGELOG.md:6]: We've already planned for the next milestone and we know its version now, so we can put in the CHANGELOG as well 🙂

```suggestion
## Unreleased: 0.8.0.0
```

**Review by chshersh — COMMENTED** [review]
@aleator Thanks for working on this! Looking great 👍 
I have a few suggestions, but they are mostly stylistic, nothing criticial.

File: src/Relude/Monad.hs [hunk 2]
```diff
@@ -47,6 +50,41 @@ chainedTo :: Monad m => (a -> m b) -> m a -> m b
...
+@
+
+bar :: IO (Async Void)
+bar = async . forever $ do ...
+
+@
+
+Both of these are well typed but 'foo' has a serious mistake which
+is not caught by the type system. In contrast, using 'infinitely'
+instead of 'forever' in 'foo' is a type error.
```
[the remark this answers is not in the document]
Comment (aleator) [reply src/Relude/Monad.hs:79]: Thanks. Sorry about the typos.

**Review by vrom911 — APPROVED** [review]
Amazing addition! ✨ 

And thank for the nice docs and HLing hints as well 👍🏼

**Pull request merged**